src/test/java/com/puppycrawl/tools/checkstyle/PackageObjectFactoryTest.java

```diff
@@ -150,6 +165,77 @@ private void testCreateModuleWithNonExistName(String name) {
     }
 
     @Test
+    public void testCreateObjectFromMap() throws Exception {
+        final Method method = factory.getClass().getDeclaredMethod(
+                "createObjectFromMap", String.class, Map.class);
+        method.setAccessible(true);
+        final String moduleName = "Foo";
+        final String name = moduleName + CHECK_SUFFIX;
+        final String fullName =
+                BASE_PACKAGE + ".packageobjectfactory.bar" + PACKAGE_SEPARATOR + name;
+        final Map<String, Set<String>> map = new HashMap<>(1);
+        map.put(name, Collections.singleton(fullName));
+
+        final Object instance1 = method.invoke(factory, name, map);
+        assertEquals(fullName, instance1.getClass().getCanonicalName());
+        final Object instance2 = method.invoke(factory, moduleName, map);
+        assertEquals(fullName, instance2.getClass().getCanonicalName());
+    }
+
+    @Test
+    public void testCreateObjectFromFullModuleNames() throws Exception {
+        final Method method = factory.getClass().getDeclaredMethod(
+                "createObjectFromFullModuleNames", String.class, Set.class);
+        method.setAccessible(true);
+        final String name = "FooCheck";
+        final String fullName =
+                BASE_PACKAGE + ".packageobjectfactory.bar" + PACKAGE_SEPARATOR + name;
+        final Object instance = method.invoke(factory, name, Collections.singleton(fullName));
+        assertEquals(fullName, instance.getClass().getCanonicalName());
+    }
+
+    @Test
+    public void testCreateObjectFromFullModuleNamesWithException() throws Exception {
+        final Method method = factory.getClass().getDeclaredMethod(
+                "createObjectFromFullModuleNames", String.class, Set.class);
+        method.setAccessible(true);
+        final String name = "FooCheck";
+        final String barFullName =
+                BASE_PACKAGE + ".packageobjectfactory.bar" + PACKAGE_SEPARATOR + name;
+        final String fooFullName =
+                BASE_PACKAGE + ".packageobjectfactory.foo" + PACKAGE_SEPARATOR + name;
+        final Set<String> fullModuleNames = new HashSet<>(Arrays.asList(barFullName, fooFullName));
+        try {
+            method.invoke(factory, name, fullModuleNames);
+            fail("Exception is expected");
+        }
+        catch (InvocationTargetException ex) {
+            final String optionalNames = fullModuleNames.stream()
+                    .reduce((fullNames, fullName) -> fullNames + STRING_SEPARATOR + fullName).get();
```

Comment:
please use there concatenation of two string, to clearly show how it should looks like and text main code by different in view test code.